### DISCUSSION TITLE
Ternary alloy KKS model (2-phase) + tensor mechanics + nucleation

### DISCUSSION THREAD

**Discussion by gka80** [idaholab/moose#22474] [Q&A Modules: Phase field] [answered]
Hello!

I am in the process of constructing a KKS model for studying the evolution of a two-phase (⍺/β) ternary system (Ti-Al-V). The simulation assumes that the orientation of the β-phase (BCC) is fixed relative to a reference (global) coordinate system and the ⍺-phase (HCP) can precipitate as 12 variants relative to this orientation. The tensor mechanics module is used to account for each of the variant eigenstrains during formation and their contribution to the global free energy. A similar implementation is found here: https://doi.org/10.1088/2053-1591/ab875a

I have tested this implementation with a few simple test cases and it appears to be working as I would expect (for the most part).

Questions:

1. I am now trying to implement nucleation and am wondering if the Discrete Nucleation System can handle the model described above?
2. I track the Al-V concentration of the β-phase and each ⍺-variant and roll that up into an auxiliary variable (i.e., xAl = xAlBeta + xAlAlpha1 + xAlAlpha2 +...) so I am not quite sure what I should do for the "DiscreteNucleation" material and the "op_names" input: Should it be 'eta1 xAlAlpha1 xVAlpha1'? And have one DiscreteNucleation material for each ⍺-variant?
3. Since the "KKSMulti..." kernels include inputs for the phase free energies (i.e., fBeta, fAlpha1, fAlpha2,...), would I then add the DiscreteNucleation materials for each ⍺-variant to the corresponding phase free energy material and include 'eta1' (or whatever variant) in the 'args'?

Hopefully all of that made sense. If anyone needs any clarification, please let me know.

Thank you!

-Garrett

**Comment by GiudGiud** [comment]
@SudiptaBiswas @gambka

**Reply by lloyd-5731** [reply]
Has there been any update on this?

**Comment by lloyd-5731** [comment]
Following

**Comment by GiudGiud** [comment]
@laagesen @dschwen

**Comment by laagesen** [comment, marked as answer]
Hi Garrett - the idea behind the discrete nucleation based approach is to add an energy penalty to the system that will force the concentration in the nucleus to the correct values for the nucleus phase, so in this case you are trying to get the Al and V concentration to their values in the alpha phase, then the order parameter values would follow once the concentration is high enough to get a stable nucleus to form. So you will want the DiscreteNucleation material with `op_names` set to the concentration variables for Al and V, and `op_values` to the equilibrium values of Al and V in the alpha phase. Then for #3 you are right on that you would add the DiscreteNucleation material to the phase free energy for each phase.

**Reply by gka80** [reply]
Hi Larry-

Perfect! Thank you for the quick response. If I have any other questions, I will reach out.

Thanks again.
